python/src/scipp/show.py

```diff
@@ -405,25 +415,87 @@ def make_svg(self):
+                    offset[1] += 0.3
+                return offset
+
+            def _append_dot(content, offset, layout_direction):
+                dot = f'<circle cx="{offset[0]+0.15}" cy="{offset[1]+1.9}" r="0.1" fill="black" />'  # noqa #501
+                content += dot
+                offset = _pad_dot(offset, layout_direction)
+                return content, offset
+
+            offset = _pad_dot(offset, layout_direction)
+            content, offset = _append_dot(content, offset, layout_direction)
+            content, offset = _append_dot(content, offset, layout_direction)
+            content, offset = _append_dot(content, offset, layout_direction)
+            _pad_dot(offset, layout_direction)
+
+            ellipsis_size = [1.5, 2.0]
+            width, height, offset = _new_size_and_offset(
+                ellipsis_size, width, height, layout_direction)
+
+            return content, width, height, offset
+
+        def draw_area(area, layout_direction, reverse=False, truncate=False):
             content = ''
             width = 0
             height = 0
             offset = [0, 0]
-            if reverse:
-                area = reversed(area)
-            for name, data, color in area:
-                drawer = VariableDrawer(data, margin, target_dims=dims)
-                content += drawer.draw(color=color, offset=offset, title=name)
-                size = drawer.size()
-                if layout_direction == 'x':
-                    width += size[0]
-                    height = max(height, size[1])
-                    offset = [width, 0]
+
+            number_of_items = len(area)
+            min_items_before_worth_truncate = 5
+            if truncate and number_of_items > min_items_before_worth_truncate:
+                if reverse:
+                    first_item = area[-1]
+                    last_item = area[0]
                 else:
-                    width = max(width, size[0])
-                    height += size[1]
-                    offset = [0, height]
+                    first_item = area[0]
+                    last_item = area[-1]
+                # Draw first variable
+                content, width, height, offset = _append_drawer(
+                    content, width, height, offset, layout_direction,
+                    first_item)
+                # Draw ellipsis
+                content, width, height, offset = _append_ellipsis(
+                    content, width, height, offset, layout_direction)
+                # Draw last variable
+                content, width, height, offset = _append_drawer(
+                    content, width, height, offset, layout_direction,
+                    last_item)
+            else:
+                if reverse:
+                    area = reversed(area)
+                for item in area:
+                    content, width, height, offset = _append_drawer(
+                        content, width, height, offset, layout_direction, item)
```

Comment:
Could this be simplified by conditionally replacing `aread[1:-1]` with the ellipses?